### PR TITLE
Cleanup test/core/pthread/create. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Mostly convert to C to match the other tests here.

I verified that the normal C operators produce the desired atomic operations (e.g. sum++ produced i32.atomic.rmw.add, etc)

See #3494

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2117,7 +2117,7 @@ def compile_source_file(i, input_file):
     if not shared.Settings.SIDE_MODULE: # shared libraries/side modules link no C libraries, need them in parent
       extra_files_to_link = system_libs.get_ports(shared.Settings)
       if '-nostdlib' not in newargs and '-nodefaultlibs' not in newargs:
-        link_as_cxx = run_via_emxx
+        link_as_cxx = run_via_emxx or shared.Settings.EMBIND
```
Comment (kripken) [inline emcc.py:2120]: looks like part of another PR maybe?

File: tests/test_core.py [hunk 2]
```diff
@@ -8129,13 +8129,11 @@ def test_pthread_exceptions(self):
...
     self.do_run_in_out_file_test('tests', 'core', 'pthread', 'emscripten_atomics.c')
 
-  @no_asan('incompatibility with atomics')
   @node_pthreads
   def test_emscripten_atomics(self):
     self.set_setting('USE_PTHREADS', '1')
     self.do_run_in_out_file_test('tests', 'core', 'pthread', 'emscripten_atomics.c')
 
-  @no_asan('incompatibility with atomics')
   @node_pthreads
```
Comment (kripken) [inline tests/test_core.py:8137]: asan only runs on chromium CI after commit, so it's not tested here, but I assume you saw it works?

**Review by kripken — APPROVED** [review]
lgtm with comments addressed.

**Comment by sbc100** [comment]
Reviving 2 year old PR!

**Pull request merged**